rs/index/src/collection/collection.rs

```diff
@@ -222,7 +222,20 @@ impl<Q: Quantizer + Clone + Send + Sync + 'static> Collection<Q> {
                                     });
                             }
                             WalOpType::Delete => {
-                                // TODO(hicder): Implement delete
+                                let doc_ids = wal_entry.doc_ids;
+                                let user_ids = wal_entry.user_ids;
+                                assert!(wal_entry.data.is_empty());
+                                let ver = versions.get(&version).unwrap();
+                                user_ids.iter().try_for_each(|&user_id| {
+                                    doc_ids.iter().try_for_each(|&doc_id| {
+                                        mutable_segment.read().invalidate(user_id, doc_id, entry_seq_no)?;
+                                        for segment_name in ver.toc.iter() {
+                                            let segment = all_segments.get(segment_name).unwrap();
+                                            segment.remove(user_id, doc_id)?;
+                                        }
```

Comment:
just call the `remove` function (also you'll need to send the invalidation to `pending_mutable_segment` as well

Reply:
hmm but at this point `pending_mutable_segment` should be `None`, since we are at restart, right?

Reply:
good point - i think if that's the case, I wonder if we can just grab mutable segment lock outside and avoid grabbing lock every iteration (for both insert and delete) hmmm